### ISSUE TITLE
Do suggest Python packages when amount of rebuilds < 10

### ISSUE THREAD

**Issue by FRidh** [ryantm/nixpkgs-update#83]
The Python package set keeps increasing in size. Your bot and @ofborg don't check the reverse dependencies. Python packages typically have complex dependency trees, which often means that if one package changes others must as well. For that reason I asked earlier not to include Python packages.

I think we can run the risk of merging updates without checking the reverse dependencies as long as the amount of required rebuilds are low, because those packages are typically less important. Therefore, I think it may be interesting to suggest Python package updates as well, as long as the amount of rebuilds is low, say < 10.

In case of non-Python packages we let it up to the Nixpkgs members to decide whether to merge or not. Because not all are aware how entangled Python packages are I prefer not to do that and only suggest updates when rebuilds are < 10.

**Comment by ryantm** [comment]
Exciting! I like this idea, and another way I could relax current restrictions would be to allow updating packages in all the cases where the number of rebuilds equals 1.

This depends on #42.

**Comment by ryantm** [comment]
#42 is done, so this work can now commence.

**Comment by ryantm** [comment]
I started running this on the python packages. Please note many packages cannot be updated because they are in the python-packages.nix file, and r-ryantm skips editing any file that has multiple fetchers in it.

Once https://github.com/NixOS/nixpkgs/pull/48633 hits the merge-base of master and staging (when staging is merged into master) this will be a lot better.

**Comment by ryantm** [comment]
@FRidh I'm going to close this for now, but hopefully you can provide some feedback about how it is working for you after it runs for a bit.

**Comment by FRidh** [comment]
Thanks. This looks to be working well. So far I haven't encountered any issues aside from the funny thing mentioned regarding the naming (https://github.com/ryantm/nixpkgs-update/issues/72).

**Comment by ryantm** [comment]
@FRidh I did a special run of only the Python packages and it is done. Here is the log file:

https://gist.github.com/ryantm/0434bbae8f5455c617259725b3fce6fc/raw

**Comment by FRidh** [comment]
Good, thanks!

It would be nice to eventually create some kind of dash board, showing all successful and attempted updates.

Orthogonal to this, though I can clearly see the relation, it would be great eventually if we have a page listing unstable and stable packages along with versions, and some way to find commits related to every package in order to give the possibility to cherry-pick.
